src/frontend/src/pages/CalendarPage/DesignReviewCreateModal.tsx

```diff
@@ -158,13 +159,30 @@ export const DesignReviewCreateModal: React.FC<DesignReviewCreateModalProps> = (
             const onClear = () => {
               setValue('wbsNum', '');
               onChange('');
+              setValue('teamTypeId', '');
             };
+
+            const handleWorkPackageSelect = async (selectedValue: string) => {
+              onChange(selectedValue);
+              setValue('wbsNum', selectedValue);
+
+              const workPackage = allWorkPackages.find((wp) => wbsPipe(wp.wbsNum) === selectedValue);
+              if (workPackage) {
+                const teamType = workPackage.teamTypes.find((type) => type !== undefined);
```

Comment:
you dont need this type !== undefined operator. fixing the backend will make this never be undefined. If you look at the types you will see you are declaring the workPackage.teamTypes is an array of TeamType, not an array of TeamType | undefined. if you are getting undefined there is something critically wrong with the code, which is what is happening in the backend with your force unwrapping of a teams team type

Reply:
you should just be getting the first team type from this array if the arrays length is greater than 0.